subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -72,16 +78,19 @@ public WatchingVirtualFileSystem(
         FileWatcherRegistryFactory watcherRegistryFactory,
         AbstractVirtualFileSystem delegate,
         DelegatingDiffCapturingUpdateFunctionDecorator delegatingUpdateFunctionDecorator,
-        Predicate<String> watchFilter
+        Predicate<String> watchFilter,
+        DaemonDocumentationIndex daemonDocumentationIndex
     ) {
         super(delegate);
         this.watcherRegistryFactory = watcherRegistryFactory;
         this.delegatingUpdateFunctionDecorator = delegatingUpdateFunctionDecorator;
         this.watchFilter = watchFilter;
+        this.daemonDocumentationIndex = daemonDocumentationIndex;
     }
 
     @Override
     public void afterBuildStarted(boolean watchingEnabled) {
+        fileWatchingException = null;
```

Comment:
Shall we call this `reasonForNotWatchingFiles` or something that describes its role instead of its type?

Reply:
Renamed.